### ISSUE TITLE
388.2 Beta 2: AX-5400 Low upload speed with Wireguard turned on.

### ISSUE THREAD

**Issue by Sp0Xik** [gnuton/asuswrt-merlin.ng#386]
**Router Model Affected**
Models: AX-5400 v.1

**Firmware Version Affected**
388.2 beta 2

**Is this bug present in upstream Merlin releases too?**
Yes (388.1 beta 1)

**Describe the bug**
Upload speed with Wireguard enabled is still limited to 250 Mbps. Although the download speed after the update became as it should (750 Mbps). At the same time, on the Merlin forum, I saw that other people's speed, not only downloads, but also uploads, increased to the required values.

**To Reproduce**
1. Enable VPN or VPN Director using Wireguard.

**Expected behavior**
I expect the upload speed to be equal to the download speed.

**Screenshots**
![image](https://github.com/gnuton/asuswrt-merlin.ng/assets/129360752/b6f15170-7c08-4a4c-a0c7-3a41cd6a3f12)
![image](https://github.com/gnuton/asuswrt-merlin.ng/assets/129360752/3c6f8cc5-dfa3-48fb-940f-04d1e2653f8d)

**Comment by gnuton** [comment]
@Sp0Xik did you tried to reset to factory?
can you show me a spedtest ookla speedtest?

BTW no need to create a new issue for it.. the previous one was fine too

**Comment by Sp0Xik** [comment]
Hey @Gnuton
Here is a speed test with Wireguard enabled:
![image](https://github.com/gnuton/asuswrt-merlin.ng/assets/129360752/fa86cfd0-d627-4c73-bdc9-e98c35fac616)

Here is a speed test without VPN:
![image](https://github.com/gnuton/asuswrt-merlin.ng/assets/129360752/54daa8b4-3ab1-410f-8d57-51bc4ca17a89)


After installing 388.2 beta 1, I reset the router to factory settings, but the problem was the same. I don't think that the dirty update on beta 2 could somehow affect this.

**Comment by gnuton** [comment]
was wireshark in 388.1 performing better?

**Comment by Sp0Xik** [comment]
No. in 388.1 when wireguard was running, NAT acceleration was disabled, from which both speeds rested at 250mbps
